src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -121,142 +116,162 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
-            foreach (var peReference in currentProject.MetadataReferences.OfType<PortableExecutableReference>())
+            // Get extension method items from PE
+            foreach (var (peReference, symbolInfo) in indices.SymbolInfos!)
             {
-                var info = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
-                    solution, peReference, loadOnly: !forceIndexCreation, cancellationToken).ConfigureAwait(false);
-
-                if (info == null)
-                {
-                    // Don't provide anything if we don't have all the required SymbolTreeInfo created.
-                    return GetIndicesResult.NoneResult;
-                }
-
-                if (info.ContainsExtensionMethod)
+                var filter = CreateAggregatedFilter(targetTypeNames, symbolInfo);
+                if (currentCompilation.GetAssemblyOrModuleSymbol(peReference) is IAssemblySymbol assembly)
                 {
-                    symbolBuilder.Add(info);
+                    var internalsVisible = currentAssembly.IsSameAssemblyOrHasFriendAccessTo(assembly);
+                    var matchingMethodSymbols = GetPotentialMatchingSymbolsFromAssembly(assembly, filter, namespaceFilter, !internalsVisible, ignoreUnspecifiedContainerAccessibility: true, ignoreUnspecifiedMethodAccessibility: true, counter, cancellationToken);
+                    GetExtensionMethodItemsWorker(position, semanticModel, receiverTypeSymbol, matchingMethodSymbols, isSymbolFromCurrentCompilation: false, builder, namespaceNameCache);
                 }
             }
 
-            var syntaxIndices = syntaxBuilder.ToImmutable();
-            var symbolInfos = symbolBuilder.ToImmutable();
-
-            return new GetIndicesResult(hasResult: true, syntaxIndices, symbolInfos);
+            return builder.ToImmutable();
         }
 
-        private static MultiDictionary<string, string> CreateAggregatedFilter(ImmutableArray<string> targetTypeNames, ImmutableArray<CacheEntry> syntaxIndices, ImmutableArray<SymbolTreeInfo> symbolInfos)
+        private static void CheckDefaultAccessibility(Compilation compilation, bool internalsVisible, out bool ignoreUnspecifiedContainerAccessibility, out bool ignoreUnspecifiedMethodAccessibility)
         {
-            var results = new MultiDictionary<string, string>();
+            switch (compilation.Language)
+            {
+                // In C#, default accessibility for type is internal and for method is private.
+                case LanguageNames.CSharp:
+                    ignoreUnspecifiedContainerAccessibility = !internalsVisible;
+                    ignoreUnspecifiedMethodAccessibility = true;
+                    break;
+
+                // In VB, extension methods are declared in modules and default to public
+                case LanguageNames.VisualBasic:
+                    ignoreUnspecifiedContainerAccessibility = false;
+                    ignoreUnspecifiedMethodAccessibility = false;
```

Comment:
i'm a bit confused by this.  do you not have symbols you can ask teh real accessibility for?

Reply:
Yes, but symbol only have declared accessibility, this is meant for a quick filter when they don't have explicit accessibility modifier.

Reply:
I was wrong, fixed